### PR TITLE
[libevent] no absolute paths

### PR DESCRIPTION
**Pull request by autoantwort**
For #20469

### CODE REVIEW

File: ports/libevent/portfile.cmake [hunk 1]
```diff
@@ -40,33 +39,21 @@ vcpkg_configure_cmake(
...
-if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
-    set(_target_suffix static)
-else()
-    set(_target_suffix shared)
-endif()
-vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/share/libevent/LibeventTargets-${_target_suffix}.cmake
-    "${CURRENT_PACKAGES_DIR}"
-    "${CURRENT_INSTALLED_DIR}"
-)
-
```
Comment (NancyLi1013) [inline ports/libevent/portfile.cmake:66]: Why do you remove this part? Is this not need for current version now?

Comment (autoantwort) [reply ports/libevent/portfile.cmake:66]: No this is not needed anymore. There is no absolute path in the config anymore.

**Comment by NancyLi1013** [comment]
`evpp `and `wangle `failed on linux and osx with the errors like this:

```
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Error at /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:788 (_find_package):
  Could not find a package configuration file provided by "Libevent" with any
  of the following names:

    LibeventConfig.cmake
    libevent-config.cmake

  Add the installation prefix of "Libevent" to CMAKE_PREFIX_PATH or set
  "Libevent_DIR" to a directory containing one of the above files.  If
  "Libevent" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:52 (find_package)
```

**Comment by autoantwort** [comment]
Depends on  #20439

**Comment by JackBoosY** [comment]
Can you please merge to master?

Thanks.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please run command `vcpkg x-add-version --all` then commit the changes.

File: ports/libevent/portfile.cmake [hunk 2]
```diff
@@ -63,6 +63,7 @@ vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/share/libevent/LibeventTargets-${_t
     "${CURRENT_PACKAGES_DIR}"
     "${CURRENT_INSTALLED_DIR}"
 )
+vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/share/libevent/LibeventConfig.cmake "${SOURCE_PATH}/include;${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/include" "")
```
Comment (JackBoosY) [inline ports/libevent/portfile.cmake:66]: What is the content of the absolute path in this file?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libevent/portfile.cmake (hunk 2 above)
Comment (autoantwort) [reply ports/libevent/portfile.cmake:66]: It sets `LIBEVENT_INCLUDE_DIRS` in an `if(false)` branch

**Review by PhoebeHui — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the absolution!

**Pull request merged**